hw05-bytecode/src/main/java/skislitsyn/utils/Ioc.java

```diff
@@ -0,0 +1,65 @@
+package skislitsyn.utils;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.HashSet;
+
+import skislitsyn.logging.Log;
+
+public class Ioc {
+    private Ioc() {
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> T createMyProxy(T obj) {
+	InvocationHandler handler = new MyInvocationHandler(obj);
+	return (T) Proxy.newProxyInstance(Ioc.class.getClassLoader(), obj.getClass().getInterfaces(), handler);
+    }
+
+    static class MyInvocationHandler implements InvocationHandler {
+	private final Object myObj;
```

Comment:
```suggestion
	private final Object obj;
```

Лучше все-таки просто `obj`. 
Все эти "my*" выглядят не солидно.